### DISCUSSION TITLE
What is the rationale for the 0 dBm voltage value in power source?

### DISCUSSION THREAD

**Discussion by csrabak** [ra3xdh/qucs_s#778] [Q&A] [answered]
The magnitude of the voltage source for the Qucs-s power source, available in the components->sources pane puts 0.316228 V as magnitude for a 0 dBm power, 50 Ω (default impedance).

The line of the exported netlist is:

```
VP1 out 0 dc 0 ac 0.316228 SIN(0 0.316228 1MEG) portnum 1 z0 50
```

Since simulators use magnitude for this parameter, which is the _peak_ voltage of the source, this value would be OK for an open circuit, but due the internal impedance only one fourth of a milliwatt is available to the matched load.

Shouldn't the voltage be such that when a matched (i.e. 50 Ω) load is fed the power delivered to the load be 0 dBm (1 mW)?

**Comment by tomhajjar** [comment]
The voltage is measured with a resistive load equal to the source resistance. Qucs/Spice uses the Peak voltage, 1.414 x RMS.

I forgot to show the data from ngspice was different than Qucsator...

![2024-06-23_084522](https://github.com/ra3xdh/qucs_s/assets/102838604/7e517210-c127-4815-9d16-3a6170028c0c)

[dBm_Voltage.zip](https://github.com/user-attachments/files/15944613/dBm_Voltage.zip)

**Comment by ra3xdh** [comment]
See #779

**Comment by ra3xdh** [comment, marked as answer]
The calculation has been fixed #780 The result from Qucsator and Ngspice are matching now. Thanks for pointing this.

**Reply by csrabak** [reply]
Your are welcome! I thank you for quick reply and fixing this!
